src/types/transaction.cairo

```diff
@@ -143,8 +143,24 @@ pub impl TransactionImpl of TransactionTrait {
     fn wtxid(self: @Transaction) -> Hash {
         double_sha256_byte_array(@(self.encode_with_witness()))
     }
-}
 
+    fn weight(self: @Transaction) -> usize {
+        let mut size: usize = 0;
+
+        size += self.version.encode().len() * NON_WITNESS_FACTOR;
```

Comment:
What might make sense is to add `byte_size` to the Encode trait, so that we don't waste bytearrays here